### PR TITLE
feat: add isLoading flag to ad package to manage placeholder

### PR DESCRIPTION
**Pull request by colincclark**
- added isLoading prop to the Ad component in order to show a placeholder to avoid page jankiness particularly in article list pages
- added tests for `ad` component but this needs more TLC (in a future PR)
- fixed `ad` component prop-types

NOTE: ads do not work in native - and never did. See https://github.com/newsuk/times-components/issues/1066

Future PR to follow will handle the empty ad state (e.g. with ad blocker) and increase code coverage and tidy up the other tests including the DOMContext tests

### CODE REVIEW

File: packages/ad/__tests__/android/__snapshots__/ad.android.test.js.snap [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+        <Text>
+          ADVERTISEMENT
+        </Text>
+      </View>
+    </View>
+  </View>,
+]
+`;
+
+exports[`3. Advert loading state placeholder 1`] = `
```
Comment (craigbilner) [inline packages/ad/__tests__/android/__snapshots__/ad.android.test.js.snap]: does this demo the loading state?

File: packages/ad/__tests__/web/__snapshots__/ad.web.test.js.snap [hunk 2]
```diff
@@ -0,0 +1,396 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`1. Advert 1`] = `
```
Comment (craigbilner) [inline packages/ad/__tests__/web/__snapshots__/ad.web.test.js.snap]: with placeholder?

File: packages/ad/__tests__/web/ad.web.test.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
+import React, { Fragment } from "react";
+import { mount } from "enzyme";
+import {
```
Comment (craigbilner) [inline packages/ad/__tests__/web/ad.web.test.js]: going to have to revisit this _improved_ DX :joy:

File: packages/ad/__tests__/android/__snapshots__/ad.android.test.js.snap (hunk 1 above)
Comment (colincclark) [reply packages/ad/__tests__/android/__snapshots__/ad.android.test.js.snap]: https://github.com/newsuk/times-components/issues/1066

File: packages/ad/__tests__/web/__snapshots__/ad.web.test.js.snap (hunk 2 above)
Comment (colincclark) [reply packages/ad/__tests__/web/__snapshots__/ad.web.test.js.snap]: https://github.com/newsuk/times-components/issues/1066

Comment (colincclark) [reply packages/ad/__tests__/web/__snapshots__/ad.web.test.js.snap]: the snaps definitely need fixing up, could do with a pair or second set of 👀 on this package

**Pull request merged**